File: src/main/java/bbarena/model/choice/Action.java
package bbarena.model.choice;

import bbarena.model.Choice;

/**
 * Created by Teg on 03/12/2016.
 */
public enum Action implements Choice {
    MOVE(Boolean.FALSE),
    BLOCK(Boolean.FALSE),
    BLITZ(Boolean.TRUE),
    FOUL(Boolean.TRUE),
    PASS(Boolean.TRUE),
    HANDOFF(Boolean.TRUE);

    Action(Boolean singleUse) {
        _singleUse = singleUse;
    }

    private final Boolean _singleUse;

    public Boolean isSingleUse() {
        return _singleUse;
    }
}

File: src/main/kotlin/bbarena/model/choice/Action.kt
package bbarena.model.choice

import bbarena.model.Choice

/**
 * Created by Teg on 03/12/2016.
 */
enum class Action(val isSingleUse: Boolean) : Choice {
    MOVE(java.lang.Boolean.FALSE),
    BLOCK(java.lang.Boolean.FALSE),
    BLITZ(java.lang.Boolean.TRUE),
    FOUL(java.lang.Boolean.TRUE),
    PASS(java.lang.Boolean.TRUE),
    HANDOFF(java.lang.Boolean.TRUE)
}
